service/debugger/debugger.go

```diff
@@ -182,16 +193,45 @@ func (d *Debugger) CreateBreakpoint(requestedBp *api.Breakpoint) (*api.Breakpoin
 	if err != nil {
 		return nil, err
 	}
-	bp.Tracepoint = requestedBp.Tracepoint
-	bp.Goroutine = requestedBp.Goroutine
-	bp.Stacktrace = requestedBp.Stacktrace
-	bp.Variables = requestedBp.Variables
-	bp.Cond = nil
+	if err := copyBreakpointInfo(bp, requestedBp); err != nil {
+		d.process.ClearBreakpoint(bp.Addr)
+		return nil, err
+	}
 	createdBp = api.ConvertBreakpoint(bp)
 	log.Printf("created breakpoint: %#v", createdBp)
 	return createdBp, nil
 }
 
+func (d *Debugger) AmendBreakpoint(amend *api.Breakpoint) error {
+	var original *proc.Breakpoint
+	for i := range d.process.Breakpoints {
```

Comment:
`FindBreakpoint` returns an `api.Breakpoint` this needs a `proc.Breakpoint`. I have refactored the code of `FindBreakpoint` and `AmendBreakpoint` to both use a `findBreakpoint` that returns a `proc.Breakpoint`, this also works as a micro-optimizations for `FindBreakpoint`.